packages/mermaid/src/diagrams/sankey/sankeyRenderer.ts

```diff
@@ -0,0 +1,204 @@
+  // That will be a function that calculates nodes and links dimensions
+  //
+  const nodeWidth = 10;
+  const sankey = d3Sankey()
+    .nodeId((d: any) => d.id) // we use 'id' property to identify node
+    .nodeWidth(nodeWidth)
+    .nodePadding(10)
+    .nodeAlign(nodeAlign)
+    .extent([
+      [0, 0],
+      [width, height],
+    ]);
+
+  // Compute the Sankey layout: calculate nodes and links positions
+  // Our `graph` object will be mutated by this and enriched with other properties
+  //
+  sankey(graph);
+
+  // Get color scheme for the graph
+  const colorScheme = d3scaleOrdinal(d3schemeTableau10);
+
+  // Create rectangles for nodes
+  svg
+    .append('g')
+    .attr('class', 'nodes')
+    .selectAll('.node')
+    .data(graph.nodes)
+    .join('g')
+    .attr('class', 'node')
+    .attr('transform', function (d: any) {
+      return 'translate(' + d.x0 + ',' + d.y0 + ')';
+    })
+    .attr('x', (d: any) => d.x0)
+    .attr('y', (d: any) => d.y0)
+    .append('rect')
+    .attr('height', (d: any) => {
+      return d.y1 - d.y0;
+    })
+    .attr('width', (d: any) => d.x1 - d.x0)
+    .attr('fill', (d: any) => colorScheme(d.id));
+
+  // Create labels for nodes
+  svg
+    .append('g')
+    .attr('class', 'node-labels')
+    .attr('font-family', 'sans-serif')
+    .attr('font-size', 14)
+    .selectAll('text')
+    .data(graph.nodes)
+    .join('text')
+    .attr('x', (d: any) => (d.x0 < width / 2 ? d.x1 + 6 : d.x0 - 6))
+    .attr('y', (d: any) => (d.y1 + d.y0) / 2)
+    .attr('dy', '0.35em')
+    .attr('text-anchor', (d: any) => (d.x0 < width / 2 ? 'start' : 'end'))
+    .text((d: any) => d.id);
+
+  // Creates the paths that represent the links.
+  const link = svg
+    .append('g')
+    .attr('class', 'links')
+    .attr('fill', 'none')
+    .attr('stroke-opacity', 0.5)
+    .selectAll('.link')
+    .data(graph.links)
+    .join('g')
+    .attr('class', 'link')
+    .style('mix-blend-mode', 'multiply');
+
+  const linkColor = conf?.linkColor || SankeyLinkColor.gradient;
+
+  if (linkColor === SankeyLinkColor.gradient) {
+    const gradient = link
+      .append('linearGradient')
+      .attr('id', (d: any) => (d.uid = Uid.next('linearGradient-')).id)
+      .attr('gradientUnits', 'userSpaceOnUse')
+      .attr('x1', (d: any) => d.source.x1)
+      .attr('x2', (d: any) => d.target.x0);
+
+    gradient
+      .append('stop')
+      .attr('offset', '0%')
+      .attr('stop-color', (d: any) => colorScheme(d.source.id));
+
+    gradient
+      .append('stop')
+      .attr('offset', '100%')
+      .attr('stop-color', (d: any) => colorScheme(d.target.id));
+  }
+
+  let coloring: any;
+  switch (linkColor) {
+    case SankeyLinkColor.gradient:
+      coloring = (d: any) => d.uid;
+      break;
+    case SankeyLinkColor.source:
+      coloring = (d: any) => d.source.id;
+      break;
+    case SankeyLinkColor.target:
+      coloring = (d: any) => d.target.id;
+      break;
+    default:
+      coloring = linkColor;
```

Comment:
We should add image tests for these linkColors too.

@Yokozuna59 Nice catch by codecov :)